File: countElements.py
class Solution:
    def countElements(self, arr: List[int]) -> int:
        dict = {}
        for int in arr:
            if int in dict:
                dict[int] = dict[int] + 1
            else:
                dict[int] = 1

        sum = 0
        for key, val in dict.items():
            if (key + 1) in dict:
                sum += val

        return sum

File: count_elements.rb
def count_elements(arr)
  dict = {}
  arr.each { |i| dict[i] = dict[i] ? dict[i] + 1 : 1 }

  sum = 0
  dict.each_pair do |key, val|
    sum += val if dict[key+1]
  end

  sum
end
